### ISSUE TITLE
filter visible service bodies based on the bot user permissions, rather than all service bodies

### ISSUE THREAD

**Issue by nigel-bmlt** [bmlt-enabled/bmlt-workflow#59]

**Comment by pjaudiomv** [comment]
any traction on this? our server has over a 100 service bodies but when using at the area level when you only have access to 1 maybe 2 service bodies it can be confusing for the user. 

Example if you have/need all the service body data you could filter on that like yap does

```php
# admin_action=get_permissions
$permissions = '{"service_body":[{"id":110,"name":"Port City Area","permissions":2},{"id":111,"name":"South Coastal Area","permissions":2}]}';
$service_bodies = json_decode(file_get_contents("https://bmlt.sezf.org/main_server/client_interface/json/?switcher=GetServiceBodies"));
$service_bodies_for_user = json_decode($permissions);

# we do this because the response is different if perms only have 1 service body vs multiple :(
if (!is_array($service_bodies_for_user->service_body)) {
    $service_bodies_for_user = array($service_bodies_for_user->service_body);
} elseif (isset($service_bodies_for_user->service_body)) {
    $service_bodies_for_user = $service_bodies_for_user->service_body;
} else {
    $service_bodies_for_user = array();
}

$enriched_service_bodies_for_user = array();
foreach ($service_bodies_for_user as $service_body_for_user) {
    foreach ($service_bodies as $service_body) {
        if (intval($service_body->id) === $service_body_for_user->id) {
            array_push($enriched_service_bodies_for_user, $service_body);
        }
    }
}

echo json_encode($enriched_service_bodies_for_user);
```


Or you could just make a request to GetServiceBodies filtering on the ones they have access to

```php
# admin_action=get_permissions
$permissions = '{"service_body":[{"id":110,"name":"Port City Area","permissions":2},{"id":111,"name":"South Coastal Area","permissions":2}]}';
$service_bodies_for_user = json_decode($permissions);

# we do this because the response is different if perms only have 1 service body vs multiple :(
if (!is_array($service_bodies_for_user->service_body)) {
    $service_bodies_for_user = array($service_bodies_for_user->service_body);
} elseif (isset($service_bodies_for_user->service_body)) {
    $service_bodies_for_user = $service_bodies_for_user->service_body;
} else {
    $service_bodies_for_user = array();
}

$service_body_filter = '';
foreach ($service_bodies_for_user as $service_body_for_user) {
    $service_body_filter .= '&services[]=' . $service_body_for_user->id;
}

echo file_get_contents("https://bmlt.sezf.org/main_server/client_interface/json/?switcher=GetServiceBodies" . $service_body_filter);

```

**Comment by nigel-bmlt** [comment]
fixed in 1.0.4
